syntax = "proto3";

package protos;



import "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";

service UserCredentialService {
  rpc GetAll (google.protobuf.Empty) returns (GetAllUserCredentialsResponse);
  rpc GetById (GetUserCredentialByIdRequest) returns (UserCredentialModel);
  rpc GetByUsername (GetUserCredentialByUsernameRequest) returns (UserCredentialModel);
}

message UserCredentialModel {
  int32 customer_id = 1;
  string username = 2;
  string password_hash = 3;
  optional google.protobuf.Timestamp updated_at = 4;
}

message GetAllUserCredentialsResponse {
  repeated UserCredentialModel user_credentials = 1;
}

message GetUserCredentialByIdRequest {
  int32 customer_id = 1;
}

message GetUserCredentialByUsernameRequest {
  string username = 1;
}
